Formula/python3.rb

```diff
@@ -310,7 +310,22 @@ def post_install
   end
 
   def xy
-    version.to_s.slice(/(3\.\d)/) || "3.6"
+    # Stable versions
+    ver = version.to_s.slice(/(3\.\d)/)
+
+    if ver.nil?
+        if File.file?("configure.ac")
+            # In Python's source tree
+            ver = File.read("configure.ac").slice(/PYTHON_VERSION, (3\.\d)/, 1)
+        else
+            # Installed packages
+            ver = (prefix/"Frameworks/Python.framework/Versions").children.first.basename.to_s
+        end
+    end
+
+    raise "Cannot determine Python version" unless ver
```

Comment:
Before ```make install``` configure.ac is available in the source tree, and after that command Frameworks/Versions/3.x folder is there. I guess if there are future breakages only tweaking paths is necessary.